tests/apiserver-proxy-tests/test.sh

```diff
@@ -9,40 +15,37 @@ getConfigFile() {
 }
 
 test(){
-
-  if [[ -z "${POD_NAME}" ]]; then
-    echo "POD_NAME not provided"
-    echo "TEST FAILED"
-    exit 1
-  fi
-
-  UUID=$(cat /proc/sys/kernel/random/uuid)
-  echo ${UUID} > "${PWD}/uuid"
-  
-  set +e
-  KUBECTL_OUT=$(kubectl exec ${POD_NAME} cat ${PWD}/uuid)
-  KUBECTL_ERR=$((kubectl exec ${POD_NAME} cat ${PWD}/uuid) 2>&1 1>/dev/null)
-  set -e
-
-  if [[ -n "${KUBECTL_ERR}" ]]; then
-  echo "kubectl exec output:"
-  echo "${KUBECTL_OUT}"
-  echo "kubectl exec error:"
-  echo "${KUBECTL_ERR}"
-
-  echo "TEST FAILED"
-  exit 1
-  fi
-
-  if [[ "${UUID}" != "${KUBECTL_OUT}" ]]; then
-    echo "KUBECTL_OUT should be ${UUID}"
-    echo "KUBECTL_OUT is ${KUBECTL_OUT}"
-    echo "TEST FAILED"
-    exit 1
-  fi
-  echo "TEST SUCCEEDED"
+    local retry=$1
+    local maxRetries=$2
+    echo "Try $retry/$maxRetries"
+
+    UUID=$(cat /proc/sys/kernel/random/uuid)
+    echo "${UUID}" > "${PWD}/uuid"
+
+    local out
+    local status
+    set +e
+    out=$(kubectl exec "${POD_NAME}" cat ${PWD}/uuid 2>&1)
+    status=$?
+    set -e
+
+    if [[ "$status" -ne 0 ]] || [[ "${UUID}" != "${out}" ]]; then
+        echo "kubectl exec error ($status):"
+        echo "${out}"
+        echo "---"
+        echo "UUID: $(cat ${PWD}/uuid)"
+        retry=$((retry+1))
```

Comment:
"retry" shouldn't be modified.
I can see a recursive function here :) We could use some good FP principles to improve a bit:
- Use immutable variables :)
- Verify "end condition" at the very beginning of the function. This keeps the main code block simpler and shorter.
- recursive step should be simply: `test "$((retry+1))" "${maxRetries}}`